src/freenet/node/updater/MainJarDependenciesChecker.java

```diff
@@ -1232,6 +1233,20 @@ boolean deploy() {
             }
         }
         
+        /** Temp file will actually be used after restarting, so we need it to not be deleted. 
+         * @throws IOException */
+        File moveTempFile() throws IOException {
+            // Best that these NOT be absolute, want relative filenames for windows updater script.
+            File parent = filename.getParentFile();
+            if(parent == null) parent = new File(".");
+            File newTemp = File.createTempFile(filename.getName(), ".new.tmp", parent);
+            if(File.separatorChar == '\\')
+                newTemp.delete(); // FIXME Needed for Windows, no actual symlink race here AFAICS.
```

Comment:
We have `FileUtil.detectedOS.isWindows` to show whether we are on Windows, no need for `separatorChar` magic.